developers_chamber/scripts/qa.py

```diff
@@ -0,0 +1,246 @@
+        Runs shell command and returns its output.
+        """
+        try:
+            return subprocess.check_output(command, shell=True).decode().strip()
+        except subprocess.CalledProcessError as ex:
+            raise QAError(str(ex), ex.output.decode())
+
+    def _run_check(self):
+        """
+        This function should implement the actual check and should raise `QAError` if failed.
+        """
+        raise NotImplementedError()
+
+    def _cleanup(self):
+        """
+        Cleans up the repo to be fresh for another check.
+        """
+        self._get_repo().git.stash('save')
+
+    def run(self):
+        """
+        Runs the check and cleanup methods.
+        """
+        self._run_check()
+        self._cleanup()
+
+
+class QACheckRunner:
+    """
+    Runs multiple checks and evaluates the results while printing a nice output.
+    """
+    def __init__(self, *checks):
+        """
+        Arguments:
+            checks: Instances of QACheck class.
+        """
+        self.checks = checks
+
+    def run(self):
+        print('-----------------------------------------------------------')
+        print_bold(' Quality Assurance Check')
+        print('-----------------------------------------------------------')
+
+        failed_checks = []
+        for i, check in enumerate(self.checks):
+            print_heading('[{}/{}] {}...'.format(i + 1, len(self.checks), check.name))
+            try:
+                check.run()
+            except QAError as ex:
+                failed_checks.append(check)
+                print_error(ex)
+                print(ex.output)
+            else:
+                print_success('OK')
+
+        print('-----------------------------------------------------------')
+
+        if failed_checks:
+            print_error('FAILURE: {} check(s) failed!'.format(len(failed_checks)))
+            exit_code = 1
+        else:
+            print_success('SUCCESS: All QA checks passed!')
+            exit_code = 0
+
+        print('-----------------------------------------------------------')
+        exit(exit_code)
+
+
+# Specific QA Checks
+# ----------------------------------------------------------------------------------------------------------------------
+
+
+class MissingMigrationsQACheck(QACheck):
+    """
+    Checks that make migrations command does not generate new migrations.
+    """
+    name = 'Check missing migrations'
+
+    def _run_check(self):
+        output = self._run_command('pydev make-migration --dry-run')
+        last_line = output.split('\n')[-1]
+        if last_line != 'No changes detected':
+            raise QAError('Found missing migration(s)!', output)
+
+
+class MigrationFilenamesQACheck(QACheck):
+    """
+    Checks that new migrations introduced in the branch have correct names.
+    """
+    name = 'Check migration filenames'
+
+    def _is_migration_file_with_wrong_name(self, path):
+        match = re.search(r'migrations\/([^\/]+)\.py$', path)
```

Comment:
The newly generated migrations include the word migration, e.g. 0006_migration.py and it would fail this check. Maybe move the regexp to settings?

Reply:
But we have rule that we allow only numbers in migrations, or? So it is ok to fail this check. That's all what this migration is about.

Reply:
OK, clarified in person, there is always `_migration` suffix.